src/components/Navigation/Navigation.tsx

```diff
@@ -0,0 +1,41 @@
+import { NavLink, useLocation } from 'react-router-dom';
+import classes from './Navigation.module.scss';
+import React, { useCallback } from 'react';
+import { IconType } from 'react-icons';
+
+interface NavigationProps {
+  navItems: { name: string; path: string; icon?: IconType }[];
+}
+
+const Navigation: React.FC<NavigationProps> = ({ navItems }) => {
+  const location = useLocation();
+
+  const isNavLinkActive = useCallback(
+    (to: string) => {
+      return location.pathname.includes(to);
+    },
+    [location]
+  );
+
+  return (
+    <nav>
+      <ul className={`${classes['c-navigation-container']}`}>
+        {navItems.map((item) => (
+          <li key={item.path}>
+            <NavLink
+              to={item.path}
+              className={`${classes['c-navigation-container__nav-link']} ${isNavLinkActive(item.path) && classes['c-navigation-container__nav-link--active']}`}
+            >
+              {item.icon &&
+                React.createElement(item.icon, {
+                  className: `${classes['c-navigation-container__nav-link-icon']} ${isNavLinkActive(item.path) && classes['c-navigation-container__nav-link-icon--active']}`
+                })}
+              <span>{item.name}</span>
+            </NavLink>
```

Comment:
There is no point of creating `<NavLink>` component if you write rest of stuff here